lib/src/codecs/text/l10n/nl.dart

```diff
@@ -0,0 +1,166 @@
+
+  @override
+  String inMonths(String months, {InOnVariant variant = InOnVariant.simple}) =>
+      '${_inVariant(variant)} $months';
+
+  @override
+  String inWeeks(String weeks, {InOnVariant variant = InOnVariant.simple}) =>
+      '${_inVariant(variant)} week $weeks';
+
+  String _inVariant(InOnVariant variant) {
+    return switch (variant) {
+      InOnVariant.simple => 'in',
+      InOnVariant.also => 'die ook in',
+      InOnVariant.instanceOf => 'van',
+    };
+  }
+
+  @override
+  String onDaysOfWeek(
+    String days, {
+    bool indicateFrequency = false,
+    DaysOfWeekFrequency? frequency = DaysOfWeekFrequency.monthly,
+    InOnVariant variant = InOnVariant.simple,
+  }) {
+    assert(variant != InOnVariant.also);
+
+    final frequencyString =
+        frequency == DaysOfWeekFrequency.monthly ? 'maand' : 'jaar';
+    final suffix = indicateFrequency ? ' van de $frequencyString' : '';
+    return '${_onVariant(variant)} $days$suffix';
+  }
+
+  @override
+  String? get weekdaysString => 'weekdagen';
+  @override
+  String get everyXDaysOfWeekPrefix => 'elke ';
+  @override
+  String nthDaysOfWeek(Iterable<int> occurrences, String daysOfWeek) {
+    if (occurrences.isEmpty) return daysOfWeek;
+
+    final ordinals = list(
+      occurrences.map(ordinal).toList(),
+      ListCombination.conjunctiveShort,
+    );
+    return 'de $ordinals $daysOfWeek';
+  }
+
+  @override
+  String onDaysOfMonth(
+    String days, {
+    DaysOfVariant daysOfVariant = DaysOfVariant.dayAndFrequency,
+    InOnVariant variant = InOnVariant.simple,
+  }) {
+    final suffix = {
+      DaysOfVariant.simple: '',
+      DaysOfVariant.day: ' dag',
+      DaysOfVariant.dayAndFrequency: ' dag van de maand',
+    }[daysOfVariant];
+    return '${_onVariant(variant)} de $days$suffix';
+  }
+
+  @override
+  String onDaysOfYear(
+    String days, {
+    InOnVariant variant = InOnVariant.simple,
+  }) =>
+      '${_onVariant(variant)} de $days dag van het jaar';
+
+  String _onVariant(InOnVariant variant) {
+    return switch (variant) {
+      InOnVariant.simple => 'op',
+      InOnVariant.also => 'die ook op',
+      InOnVariant.instanceOf => 'van',
+    };
+  }
+
+  @override
+  String list(List<String> items, ListCombination combination) {
+    final (two, end) = switch (combination) {
+      ListCombination.conjunctiveShort => (' & ', ' & '),
+      ListCombination.conjunctiveLong => (' en ', ', en '),
+      ListCombination.disjunctive => (' of ', ', of '),
+    };
+    return RruleL10n.defaultList(items, two: two, end: end);
+  }
+
+  @override
+  String ordinal(int number) {
+    assert(number != 0);
+    if (number == -1) return 'laatste';
+
+    final n = number.abs();
+    String string;
+    if (n % 10 == 1 && n % 100 != 11) {
+      string = '${n}ste';
+    } else if (n % 10 == 2 && n % 100 != 12) {
+      string = '${n}de';
+    } else if (n % 10 == 3 && n % 100 != 13) {
+      string = '${n}de';
+    } else {
+      string = '${n}de';
+    }
```

Comment:
Do you have an example where ordinals are formatted like this? I can only find examples that always append an “e”. For example, in Unicode's [plural rules](https://www.unicode.org/cldr/charts/43/supplemental/language_plural_rules.html#nl) and in [Common Locale Data Repository](https://github.com/unicode-org/cldr/blob/9bbbc7769d6824229c49817c143ca7afebb00a34/common/main/nl.xml#L5119-L5124).

```suggestion
    final string = '${number.abs()}e';
```

Reply:
I made some changes before your comment and we have 2 options for the ordinals in Dutch:
1ste, 2de, 3de,...  or 1e, 2e, 3e,...
I've picked your suggestion as it is smaller in code.